[ATO-1122] Add response_ids to domain (#12466)
RasaHQ/rasa: docs/docs/responses.mdx

In this example, when `utter_greet` gets predicted as the next action, Rasa will randomly pick one 
of the two response variations to use.

> #### IDs for Responses

Reviewer: This entire section needs to go under L46 🙏🏻

Author: Why, when response variations are only introduced at L91?
It does not make sense to provide example with multiple responses before we defined them.

In this example, when `utter_greet` gets predicted as the next action, Rasa will randomly pick one 
of the two response variations to use.

> #### IDs for Response Variations

Reviewer: Sorry to be a pain, but the ids can be used for any response, not only CRVs, so we need a section that is generalizable, so I'd make the below change and move it up after defining responses section.
```suggestion
### IDs for Responses
```

:::note New in Rasa 3.6

> You can now set ID for a response variation.
> This is useful when you want to use NLG server to create
> a specific response for a specific channel based on the data from the tracker (slots).

Reviewer: ```suggestion
You can now set ID for any response.
This is useful when you want to use the [NLG server](./nlg.mdx) to create
a specific response for a specific channel based on the data from the tracker (slots).
```

Let's add a link to the nlg page.

Reviewer: Actually, let's also keep it even shorter:
```suggestion
You can now set an ID for any response.
This is useful when you want to use the [NLG server](./nlg.mdx) to generate the response.
```
